### ISSUE TITLE
com.intellij.ide.startup.ServiceNotReadyException again

### ISSUE THREAD

**Issue by ofry** [Camelcade/Perl5-IDEA#2018]
```
IntelliJ IDEA 2019.1 (Ultimate Edition)
Build #IU-191.6183.87, built on March 27, 2019

JRE: 1.8.0_202-release-1483-b39 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
KUbuntu 18.10 x64
```

Perl v2019.1

```
java.lang.Exception: com.intellij.ide.startup.ServiceNotReadyException
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivity(StartupManagerImpl.java:363)
	at com.intellij.ide.startup.impl.StartupManagerImpl.lambda$queueSmartModeActivity$3(StartupManagerImpl.java:166)
	at com.intellij.openapi.project.DumbServiceImpl.lambda$runWhenSmart$0(DumbServiceImpl.java:232)
	at com.intellij.ide.startup.impl.StartupManagerImpl.lambda$runWhenProjectIsInitialized$6(StartupManagerImpl.java:392)
	at com.intellij.ui.GuiUtils.invokeLaterIfNeeded(GuiUtils.java:377)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runWhenProjectIsInitialized(StartupManagerImpl.java:394)
	at com.intellij.openapi.project.DumbServiceImpl.runWhenSmart(DumbServiceImpl.java:224)
	at com.intellij.ide.startup.impl.StartupManagerImpl.queueSmartModeActivity(StartupManagerImpl.java:166)
	at com.intellij.ide.startup.impl.StartupManagerImpl.access$400(StartupManagerImpl.java:48)
	at com.intellij.ide.startup.impl.StartupManagerImpl$1.lambda$run$0(StartupManagerImpl.java:196)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at com.intellij.ide.startup.impl.StartupManagerImpl$1.run(StartupManagerImpl.java:196)
	at com.intellij.openapi.project.DumbServiceImpl.lambda$runWhenSmart$0(DumbServiceImpl.java:232)
	at com.intellij.ide.startup.impl.StartupManagerImpl.lambda$runWhenProjectIsInitialized$6(StartupManagerImpl.java:392)
	at com.intellij.ui.GuiUtils.invokeLaterIfNeeded(GuiUtils.java:377)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runWhenProjectIsInitialized(StartupManagerImpl.java:394)
	at com.intellij.openapi.project.DumbServiceImpl.runWhenSmart(DumbServiceImpl.java:224)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runPostStartupActivities(StartupManagerImpl.java:184)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.lambda$null$7(ProjectManagerImpl.java:411)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:315)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:435)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:419)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:403)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:747)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:696)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: com.intellij.ide.startup.ServiceNotReadyException
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:741)
	at com.intellij.util.indexing.FileBasedIndexImpl.ensureUpToDate(FileBasedIndexImpl.java:709)
	at com.intellij.psi.stubs.StubIndexImpl.processAllKeys(StubIndexImpl.java:388)
	at com.intellij.psi.stubs.StubIndex.processAllKeys(StubIndex.java:82)
	at com.intellij.psi.stubs.StubIndexImpl.getAllKeys(StubIndexImpl.java:377)
	at com.perl5.lang.perl.util.PerlSubUtil.getDeclaredSubsNames(PerlSubUtil.java:188)
	at com.perl5.lang.perl.idea.project.PerlNamesCache$1.lambda$run$0(PerlNamesCache.java:46)
	at com.intellij.openapi.project.DumbService.runReadActionInSmartMode(DumbService.java:132)
	at com.perl5.lang.perl.idea.project.PerlNamesCache$1.run(PerlNamesCache.java:44)
	at com.perl5.lang.perl.idea.project.PerlNamesCache.forceCacheUpdate(PerlNamesCache.java:76)
	at com.perl5.lang.perl.idea.project.Perl5ProjectComponent.lambda$projectOpened$1(Perl5ProjectComponent.java:82)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivity(StartupManagerImpl.java:360)
	... 39 more
```

**Comment by hurricup** [comment]
Does this happen all the time or just sometimes?

**Comment by ofry** [comment]
Just sometimes.

**Comment by hurricup** [comment]
ok. will try to workaround this. the cause https://github.com/JetBrains/intellij-community/commit/e0247fda8f019c2fd5598b8da8620a79ae072e89

**Comment by hurricup** [comment]
I've published 1.1, please check

**Comment by ofry** [comment]
Thanks for solving :)
